### DISCUSSION TITLE
How can we provide secure server access without SSH keys and AWS console access?

### DISCUSSION THREAD

**Discussion by Arrayly** [gravitational/teleport#23694] [Q&A] [answered]
What solutions can be implemented, using your product to provide users secure access to AWS EC2 servers on AWS without the need for SSH keys and AWS console access? How can fine-grained permission control be ensured while setting up this solution across multiple instances?

**Comment by webvictim** [comment, marked as answer]
You could set up a Teleport cluster and use AWS EC2 auto-discovery to automatically install the Teleport SSH service on each of your instances: https://goteleport.com/docs/server-access/guides/ec2-discovery/

Teleport uses short-lived certificates and mutual TLS for all access, eliminating the need for SSH keys.

Teleport uses a label-based RBAC system for controlling access to nodes: https://goteleport.com/docs/server-access/rbac/
